xarray/core/accessor_str.py

```diff
@@ -846,68 +1452,881 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+
+            if pat.groups == 1:
+                for imatch, match in enumerate(matches):
+                    res[imatch, 0] = match
+            else:
+                for imatch, match in enumerate(matches):
+                    for jmatch, submatch in enumerate(match):
+                        res[imatch, jmatch] = submatch
+
+            return res
+
+        return self._apply(
+            # dtype MUST be object or strings can be truncated
+            # See: https://github.com/numpy/numpy/issues/8352
+            _get_res,
+            dtype=np.object_,
+            output_core_dims=[[group_dim, match_dim]],
+            output_sizes={group_dim: pat.groups, match_dim: maxcount},
+        ).astype(self._obj.dtype.kind)
+
+    def findall(
+        self,
+        pat: Union[str, bytes, Pattern],
+        case: bool = None,
+        flags: int = 0,
+    ) -> Any:
+        """
+        Find all occurrences of pattern or regular expression in the DataArray.
+
+        Equivalent to applying re.findall() to all the elements in the DataArray.
+        Results in an object array of lists.
+        If there is only one capture group, the lists will be a sequence of matches.
+        If there are multiple capture groups, the lists will be a sequence of lists,
+        each of which contains a sequence of matches.
+
+        Parameters
+        ----------
+        pat : str or re.Pattern
+            A string containing a regular expression or a compiled regular
+            expression object.
+        case : bool, default: True
+            If True, case sensitive.
+            Cannot be set if `pat` is a compiled regex.
+            Equivalent to setting the `re.IGNORECASE` flag.
+        flags : int, default: 0
+            Flags to pass through to the re module, e.g. `re.IGNORECASE`.
+            see `compilation-flags <https://docs.python.org/3/howto/regex.html#compilation-flags>`_.
+            ``0`` means no flags.  Flags can be combined with the bitwise or operator `|`.
+            Cannot be set if `pat` is a compiled regex.
+
+        Returns
+        -------
+        extracted : object array
+
+        Raises
+        ------
+        ValueError
+            `pat` has no capture groups.
+        ValueError
+            `case` is set when `pat` is a compiled regular expression.
+
+        Examples
+        --------
+        Create a string array
+
+        >>> value = xr.DataArray(
+        ...     [
+        ...         [
+        ...             "a_Xy_0",
+        ...             "ab_xY_10-bab_Xy_110-baab_Xy_1100",
+        ...             "abc_Xy_01-cbc_Xy_2210",
+        ...         ],
+        ...         [
+        ...             "abcd_Xy_-dcd_Xy_33210-dccd_Xy_332210",
+        ...             "",
+        ...             "abcdef_Xy_101-fef_Xy_5543210",
+        ...         ],
+        ...     ],
+        ...     dims=["X", "Y"],
+        ... )
+
+        Extract matches
+
+        >>> value.str.findall(r"(\\w+)_Xy_(\\d*)")
```

Comment:
I think the `\\` turns to `\` when it is rendered.

Reply:
the docstring will evaluate the escapes on import and the doctest would evaluate the result, which would require `\\\\w` for normal strings. This seems a bit confusing and makes copying the example difficult, though, so maybe you should mark the whole docstring as "raw", too: `r"""docstring"""`, which would allow us to have the docstring and the rendered result look the same.

Reply:
Okay, done.